google-cloud-spanner/src/test/resources/com/google/cloud/spanner/connection/postgresql/SetStatementTimeoutTest.sql

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+-- Test valid values
+-- Null (no timeout)
+SET STATEMENT_TIMEOUT=default;
+@EXPECT RESULT_SET 'STATEMENT_TIMEOUT','0'
+SHOW VARIABLE STATEMENT_TIMEOUT;
+
+-- Seconds
+SET STATEMENT_TIMEOUT='1s';
+@EXPECT RESULT_SET 'STATEMENT_TIMEOUT','1s'
+SHOW VARIABLE STATEMENT_TIMEOUT;
+SET STATEMENT_TIMEOUT = '2s';
+@EXPECT RESULT_SET 'STATEMENT_TIMEOUT','2s'
+SHOW VARIABLE STATEMENT_TIMEOUT;
+
+SET STATEMENT_TIMEOUT='1S';
+@EXPECT RESULT_SET 'STATEMENT_TIMEOUT','1s'
+SHOW VARIABLE STATEMENT_TIMEOUT;
+
+-- Milliseconds
+SET STATEMENT_TIMEOUT='1ms';
+@EXPECT RESULT_SET 'STATEMENT_TIMEOUT','1ms'
+SHOW VARIABLE STATEMENT_TIMEOUT;
+
+SET STATEMENT_TIMEOUT='1Ms';
+@EXPECT RESULT_SET 'STATEMENT_TIMEOUT','1ms'
+SHOW VARIABLE STATEMENT_TIMEOUT;
+
+SET STATEMENT_TIMEOUT='1mS';
+@EXPECT RESULT_SET 'STATEMENT_TIMEOUT','1ms'
+SHOW VARIABLE STATEMENT_TIMEOUT;
+
+SET STATEMENT_TIMEOUT='1MS';
+@EXPECT RESULT_SET 'STATEMENT_TIMEOUT','1ms'
+SHOW VARIABLE STATEMENT_TIMEOUT;
+
+-- Microseconds
+SET STATEMENT_TIMEOUT='1us';
+@EXPECT RESULT_SET 'STATEMENT_TIMEOUT','1us'
+SHOW VARIABLE STATEMENT_TIMEOUT;
+
+SET STATEMENT_TIMEOUT='1Us';
+@EXPECT RESULT_SET 'STATEMENT_TIMEOUT','1us'
+SHOW VARIABLE STATEMENT_TIMEOUT;
+
+SET STATEMENT_TIMEOUT='1uS';
+@EXPECT RESULT_SET 'STATEMENT_TIMEOUT','1us'
+SHOW VARIABLE STATEMENT_TIMEOUT;
+
+SET STATEMENT_TIMEOUT='1US';
+@EXPECT RESULT_SET 'STATEMENT_TIMEOUT','1us'
+SHOW VARIABLE STATEMENT_TIMEOUT;
+
+-- Nanoseconds
+SET STATEMENT_TIMEOUT='1ns';
+@EXPECT RESULT_SET 'STATEMENT_TIMEOUT','1ns'
+SHOW VARIABLE STATEMENT_TIMEOUT;
+
+SET STATEMENT_TIMEOUT='1Ns';
+@EXPECT RESULT_SET 'STATEMENT_TIMEOUT','1ns'
+SHOW VARIABLE STATEMENT_TIMEOUT;
+
+SET STATEMENT_TIMEOUT='1nS';
+@EXPECT RESULT_SET 'STATEMENT_TIMEOUT','1ns'
+SHOW VARIABLE STATEMENT_TIMEOUT;
+
+SET STATEMENT_TIMEOUT='1NS';
+@EXPECT RESULT_SET 'STATEMENT_TIMEOUT','1ns'
+SHOW VARIABLE STATEMENT_TIMEOUT;
+
+-- Test flip to higher time unit
```

Comment:
nit: I think we should rearrange this section to be in increasing order of `STATEMENT_TIMEOUT` .

Example: `SET STATEMENT_TIMEOUT='1000000ns` should be before `SET STATEMENT_TIMEOUT='1000ms'`

Reply:
makes sense